### PR TITLE
[WG proposal] Configuration endpoints

### PR DESCRIPTION
**Pull request by mpetrunic**
I diverged from original proposal on following items:
- `fork_schedule` returns array {epoch, version} object in sheet, I've changed it to `Fork` object from spec which additionally includes previous_version (I can convert back to original if we don't need previous version)
- `deposit_contract` endpoints is suppose to return contract address and chain id - I don't think we are using chain id so witched to genesis fork version so this endpoint provides all details for creating deposit (we could add min deposit amount as well?)
- Don't think it makes sense to specifie every spec param so I just put link to https://github.com/ethereum/eth2.0-specs/blob/v0.11.1/configs/mainnet.yaml

### CODE REVIEW

**Comment by mpetrunic** [comment]
from @mcdee :
  - doesn't make a lot of difference if we use the Fork object in fork schedule (I don't think the call will use previous_version, but using a standard object is a good thing); may as well stick with Fork 
  - chain ID for deposit contract is useful in a couple of situations.  First, it allows ease of use for testnets.  Second, if there is ever a contentious hard fork on Ethererum 1 we'll know which fork to follow
  - the general "configuration parameters" endpoint probably needs better definition.  Need to know how we're returning domains, for example (int, byte array, string?)

**Comment by mpetrunic** [comment]
> chain ID for deposit contract is useful in a couple of situations. First, it allows ease of use for testnets. Second, if there is ever a contentious hard fork on Ethererum 1 we'll know which fork to follow

I thought we are using fork versions instead of chain id-s. But what's the point of that in deposit contract, this probably make sense for spec/chain config endpoint. Even if there are hard forks, they will still have same deposit contract address and genesis_fork_version. Maybe we should have separate chain info endpoint or something?

> the general "configuration parameters" endpoint probably needs better definition. Need to know how we're returning domains, for example (int, byte array, string?)

They should be returned same way as that document though.
Are you suggesting we define each of them in api spec again? I would love that, but it feels like it will be difficult to maintain.

**Comment by mcdee** [comment]
> I thought we are using fork versions instead of chain id-s

Perhaps we're at cross purposes.  The chain ID is the Ethereum _1_ chain ID.  The idea is that if someone wants to make a deposit they can query the Ethereum 2 node to obtain the full details of the Ethereum 1 contract to which they should send their Ether.

> They should be returned same way as that document.

Was thinking about generic formats here.  Something like:
  - any value starting with `0x` in the spec is returned as a hex string
  - other values are returned as a number (are we okay doing this?  Are we ever likely to hit anything that could approach 2^53?)

**Comment by mpetrunic** [comment]
> Perhaps we're at cross purposes. The chain ID is the Ethereum 1 chain ID. The idea is that if someone wants to make a deposit they can query the Ethereum 2 node to obtain the full details of the Ethereum 1 contract to which they should send their Ether.

Oh, I feel so dumb now :rofl: 
Yeah eth1 chain id definitely makes sense, should I also keep genesis_fork_version as well?

**Comment by mcdee** [comment]
>  ...should I also keep genesis_fork_version as well?

To create a deposit we need the genesis fork version and the deposit domain.  Probably makes sense to have both or neither; not bothered which way we go (having them is fewer calls but duplicated data, as the same data is in the spec params).

**Comment by mpetrunic** [comment]
I would really like to spec all params individually as it could be quite helpful for people interacting with api.
 I think we should open issue for that after this is merged

**Comment by djrtwo** [comment]
In fact, deposit chain id should probably be added to eth2 configs

File: beacon-node-oapi.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
 info:
   title: "Beacon Node API"
   description: "API specification for the beacon node, which enables users to query and participate in Ethereum 2.0 phase 0 beacon chain."
   version: "0.11.1"
   contact:
-    name: EF
+    name: Ethereum Foundation
     url: https://ethereum.org
     email: "dont-contact@ethereum.org"
```
[the remark this answers is not in the document]
Comment (mpetrunic) [reply beacon-node-oapi.yaml:10]: 😂 

I have one branch where I disable lint rule for email, added better theme etc

**Pull request merged**